templates/kubernetes/terraform/modules/kubernetes/notification_service.tf

```diff
@@ -62,14 +48,15 @@ resource "helm_release" "notification_service" {
     value = var.notification_service_highly_available ? "4" : "2"
   }
 
-  # These will become secrets provided as env vars
-  set_sensitive {
-    name  = "application.sendgridApiKey"
-    value = var.notification_service_enabled && var.notification_service_sendgrid_enabled ? data.aws_secretsmanager_secret_version.sendgrid_api_key[0].secret_string : ""
+  set {
+    name  = "externalSecret.dataFrom[0]"
+    value = local.notification_service_secret_name
   }
 
-  set_sensitive {
-    name  = "application.slackApiKey"
-    value = var.notification_service_enabled && var.notification_service_slack_enabled ? data.aws_secretsmanager_secret_version.slack_api_key[0].secret_string : ""
+  set {
+    name  = "application.twilioPhoneNumber"
+    value = var.notification_service_twilio_phone_number
   }
+
+  depends_on = [data.aws_secretsmanager_secret.notification_service_secret]
```

Comment:
I think this will be a syntax error because you have a count in the secret reference, so it will need to be `data.aws_secretsmanager_secret.notification_service_secret[0]`

Reply:
Also this dependency doesn't really make sense. Typically you use `depends_on` with a resource because then you will wait until that resource exists. In the case of `data` a dependency check is not necessary because if the thing doesn't exist your terraform will fail at line 6 in this file.

Reply:
because both have them have `count` based on `notification_service_enabled` so in both `enabled` and `disabled` they wont error out, but in the case someone has it enabled but don't have the secret then it would fail. 

I wanted to surface the error earlier so it doesn't need to fail at the deployment step, I guess having it fail there is okay, will remove.

Reply:
You could also leave in the data reference but then remove the dependency but then it would still be failing at the terraform apply which has its pros and cons.. It's closer to the actual failure so potentially easier for us to give timely feedback, but erroring out and re-running the terraform isn't a great experience.

Reply:
I think its fine removing, a fresh install shouldn't run into the missing secret at all, in the rare case that happens I guess they would be able to dig into why `helm deploy` has failed